[project]
name = "pdf2md"
version = "0.1.0"
description = "A tool to convert PDF files to Markdown using various AI models"
authors = [{ name = "Thomas Kaltenbrunner", email = "" }]

requires-python = ">=3.9"

dependencies = [
    "anthropic>=0.49.0",
    "google-genai>=1.5.0",
    "markdownify>=1.1.0",
    "mistralai>=1.6.0",
    "ollama>=0.4.7",
    "openai>=1.66.3",
    "pdf2image>=1.17.0",
    "unstructured>=0.17.2",
    "unstructured-client>=0.31.5",
    "python-dotenv>=1.0.1",
    "requests>=2.32.3",
    "boto3>=1.34.0",
    "amazon-textract-textractor[pdfium]>=1.9.1",
    "pymupdf>=1.25.4",
    "tenacity>=9.0.0",
    "rouge>=1.0.1",
]

[dependency-groups]
dev = ["black>=25.1.0", "mypy>=1.15.0", "pylint>=3.3.6"]
eval = [
    "python-levenshtein>=0.27.1",
    "pandas>=2.2.3",
    "scikit-learn>=1.6.1",
    "matplotlib>=3.9.4",
    "seaborn>=0.13.2",
]

[tool.pylint.messages.control]
disable = [
    # Convention (C) messages
    "C0103", # Invalid name
    "C0114", # Missing module docstring

    # Refactor (R) messages
    "R0101", # Simplifiable if statement
    "R0903", # Too few public methods
    "R0913", # Too many positional arguments
    "R1732", # Consider using with
    "R0801", # Similar lines in multiple files

    # Warning (W) messages
    "W0102", # Dangerous default value
    "W0511", # Unspecified encoding and Fixme
    "W0601", # Global statement
    "W0611", # Unused argument and Unused import
    "W0621", # Redefining name from outer scope
    "W0703", # Catching too general exception
    "W1202", # Use str.format() for logging
    "W1401", # Anomalous backslash in string

    # Error (E) messages
    "E0110", # Abstract class instantiated
    "E0401", # Unable to import
    "E1136", # Value is unsubscriptable
]

[tool.pylint.messages.ignore-paths]
ignore-paths = [".venv"]

[tool.pylint.format]
max-line-length = 100

[tool.pylint.master]
persistent = false

[tool.pylint.design]
max-args = 10
max-attributes = 10
max-locals = 20

[tool.pylint.typecheck]
# List of members which are set dynamically and missed by Pylint inference
# system, and so shouldn't trigger E1101 when accessed.
generated-members = ["numpy.*"]

[tool.black]
line-length = 100
